neo/Network/P2P/TaskManager.cs

```diff
@@ -40,7 +41,8 @@ private void OnHeaderTaskCompleted()
         {
             if (!sessions.TryGetValue(Sender, out TaskSession session))
```

Comment:
Should `DecrementGlobalTask()` be placed before this line? Otherwise, if `session` is not found, `DecrementGlobalTask()` will never be executed.

Reply:
Well the TaskManager actor is watching for the termination of the RemoteNode, so if the session wasn’t found wouldn’t it have to have executed `OnTerminated` for the session? If so and the session had a task with a header, the GlobalTask would be decremented there. So it seems like it should be safe.